lib/io/byte-order.toit

```diff
@@ -26,6 +26,41 @@ A byte order class with support for common read and write operations on byte
   arrays.
 */
 abstract class ByteOrder:
+  /**
+  Swaps the byte-order of all 16-bit integers in the $byte-array.
+  If the integers were in little-endian order they then are in big-endian byte order
+  If the integers were in big-endian order they then are in little-endian byte order.
+  */
+  swap-16 byte-array/ByteArray -> none:
+    if byte-array.size <= 8:
+      for i := 0; i < byte-array.size; i += 2:
+        value := LITTLE-ENDIAN.uint16 byte-array i
+        BIG-ENDIAN.put-uint16 byte-array i value
+      return
+    tmp := ByteArray
+      max byte-array.size 512
```

Comment:
```suggestion
    tmp := ByteArray
        max byte-array.size 512
```

Reply:
Will do this in a follow-up PR. Here I just move the functions.